scipy/signal/_peak_finding.py

```diff
@@ -257,24 +257,39 @@ def peak_prominences(vector, peaks, wlen=None):
         Indices of peaks in `vector`.
     wlen : int, optional
         A window lenght in samples that limits the search for the lowest
-        contour line two a symmetric interval around the evaluated peak. If
-        not given the entire vector is used. Use this parameter to speed up
-        the calculation significantly for large vectors.
+        contour line two a symmetric interval around the evaluated peak. If not
+        given the entire vector is used. Use this parameter to speed up the
+        calculation significantly for large vectors.
+    ret_bases : bool
```

Comment:
3 more letters and we have full readability `return_bases`. Otherwise `compute_bases` feels like more scipy-ish. Also note that you are already allocating these arrays so there is nothing gained from not returning them. Had you not even allocated or touched any of those arrays then maybe for very large problems it would have made a change.

Now it's just a variable number of outputs ala matlab which drove some of us crazy. Instead you might want to just always return these and remove the bool.

Reply:
Okay, I can see why this is a code smell. When thinking about it the only disadvantage / annoyance I can think of is in when a function returns many values and the user is only interested in one. In Python 2.7 the syntax

    widths, *_ = peak_widths(...)

seems to be unsupported which would force the "ugly" `peak_widths()[0]` (maybe that's only me). We would definitely have to write the examples this way (at least as long as Scipy supports 2.7). 

Anyway, I think I will remove the `ret_...` handles and just use a constant output size. 

For `peak_finding` it doesn't matter anyway. Instead the current `properties` argument could force the function to calculate all possible properties even if no filter options are supplied. However this would convolute the `if`-logic a bit.

Reply:
FWIW `git grep ")\[0\]" | wc -l` on the SciPy codebase gives 399 results (and another codebase I work on with ~200k lines gives 639 results), so I think it's a sufficiently common pattern to expect users to do it :)

Reply:
Alright, that is a convincing argument. :D